pkg/xds/translate.go

```diff
@@ -40,8 +41,186 @@ const (
 	HeaderScheme    = ":scheme"
 )
 
-// ClusterName specifies cluster name for a destination
-type ClusterName func(*routingv2.Destination) string
+const (
+	// UnresolvedCluster for destinations pointing to unknown clusters.
+	UnresolvedCluster = "unresolved-cluster"
+
+	// DefaultOperation is the default decorator
+	DefaultOperation = "default-operation"
+)
+
+// ServiceByName claims a service entry from the registry using a host name.
+type ServiceByName func(host string, contextNamespace string) *model.Service
+
+// SubsetSelector resolves a subset to labels.
+type SubsetSelector func(service *model.Service, subset string) map[string]string
+
+// GuardedHost is a context-dependent virtual host entry with guarded routes.
+type GuardedHost struct {
+	// Port is the capture port (e.g. service port)
+	Port int
+
+	// Services are the services matching the virtual host.
+	// The service host names need to be contextualized by the source.
+	Services []*model.Service
+
+	// Hosts is a list of alternative literal host names for the host.
+	Hosts []string
+
+	// Routes in the virtual host
+	Routes []GuardedRoute
```

Comment:
RouteWrapper or just simply EnvoyRoute

Reply:
No, I want to keep it guarded. It's just a regular route with some extra pre-conditions that need to be checked for each proxy node.

Reply:
IMO the `Guarded_` names are fine.